Cesium3DTiles/include/Cesium3DTiles/Cesium3DTiles.h

```diff
@@ -0,0 +1,199 @@
+  static inline constexpr const char* TypeName = "Asset";
+
+  std::string version;
+
+  std::optional<std::string> tilesetVersion;
+};
+
+struct CODEGEN_API BoundingVolume : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "BoundingVolume";
+
+  std::optional<std::vector<double>> box;
+
+  std::optional<std::vector<double>> region;
+
+  std::optional<std::vector<double>> sphere;
+};
+
+struct CODEGEN_API TileContent : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "TileContent";
+
+  std::optional<BoundingVolume> boundingVolume;
+
+  std::string uri;
+};
+
+struct CODEGEN_API Tile : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "Tile";
+
+  enum class Refine { ADD, REPLACE };
+
+  BoundingVolume boundingVolume;
+
+  std::optional<BoundingVolume> viewerRequestVolume;
+
+  double geometricError;
+
+  std::optional<Refine> refine;
+
+  std::vector<double> transform =
+      {1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1};
+
+  std::optional<TileContent> content;
+
+  std::optional<std::vector<Tile>> children;
+};
+
+struct CODEGEN_API TilesetProperties : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "TilesetProperties";
+
+  double maximum;
+
+  double minimum;
+};
+
+struct CODEGEN_API Tileset : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "Tileset";
+
+  struct CODEGEN_API Properties : public CesiumUtility::ExtensibleObject {
+    static inline constexpr const char* TypeName = "Properties";
+
+    std::unordered_map<std::string, TilesetProperties> additionalProperties;
+  };
+
+  Asset asset;
+
+  std::optional<Properties> properties;
+
+  double geometricError;
+
+  Tile root;
+
+  std::optional<std::vector<std::string>> extensionsUsed;
+
+  std::optional<std::vector<std::string>> extensionsRequired;
+};
+
+struct CODEGEN_API FeatureTable : public CesiumUtility::ExtensibleObject {
+  static inline constexpr const char* TypeName = "FeatureTable";
+
+  struct CODEGEN_API BinaryBodyReference
+      : public CesiumUtility::ExtensibleObject {
+    static inline constexpr const char* TypeName = "BinaryBodyReference";
+
+    enum class ComponentType {
+      BYTE,
+      UNSIGNED_BYTE,
+      SHORT,
+      UNSIGNED_SHORT,
+      INT,
+      UNSIGNED_INT,
+      FLOAT,
+      DOUBLE
+    };
+
+    double byteOffset;
+
+    std::optional<ComponentType> componentType;
+  };
+
+  struct CODEGEN_API Property {
+    static inline constexpr const char* TypeName = "Property";
+
+    std::optional<BinaryBodyReference> v0;
+    std::optional<std::vector<double>> v1;
+    std::optional<double> v2;
+  };
+  struct CODEGEN_API GlobalPropertyScalar {
+    static inline constexpr const char* TypeName = "GlobalPropertyScalar";
+
+    struct CODEGEN_API Variant0 : public CesiumUtility::ExtensibleObject {
```

Comment:
I kinda know why we use Variant0 naming here since schema doesn't define the name for it. But it still not a good word for API alone

Reply:
Agreed but unsure how best to address this - looks like it would be fairly not-easy to override this in generator config